Fix deprecated warnings for CakePHP 3.7

stripthis opened the pull request:
Fix deprecated warnings for CakePHP 3.7 and skip log setConfig in bootstrap to avoid error if already defined.

stripthis committed:
Fix deprecated warnings for CakePHP 3.7

stripthis committed:
Skip log set config in bootstrap if already defined

ravage84 reviewed (changes requested):
Looks good.
But we should bump the version for the CakePHP dependency to 3.7.0:
https://github.com/burzum/cakephp-file-storage/blob/c0b3eefbe1d0a98fda5e9124f6404d0f64b8a7c2/composer.json#L23

burzum commented:
@ravage84 @stripthis I appreciate the PR but bumping it to 3.7 would require a new major release and not adding this to 2.x.

@stripthis can we change it so we have a conditional check and can support both version within the 2.x release?

burzum commented on src/Storage/Listener/LegacyImageProcessingListener.php:
```diff
@@ -221,15 +221,16 @@ protected function _removeVersions(Event $Event) {
 	public function afterDelete(Event $Event) {
 		if ($this->_checkEvent($Event)) {
-			$record = $Event->data['record'];
+			$record =  $Event->getData('record');
```
There is one space to much in front of the $Event

ravage84 committed:
Fix CS

ravage84 committed:
Add PHP 7.3 to test suite

ravage84 committed:
Update to CakePHP 3.7.5

ravage84 committed:
Simplify path extraction

ravage84 committed:
Comment out full if clause, not just body

ravage84 committed:
Remove unnecessary plugin loading of Burzum/Imagine

ravage84 committed:
Add backwards compatibility to CakePHP lower than 3.7.0

ravage84 committed:
Only load the plugin during test suite bootstrap when lower than CakePHP 3.7

That's BC for < 3.7.0 but now we need to load when CakePHP is newer.

ravage84 committed:
Add empty Plugin class

ravage84 committed:
Load Plugin for CakePHP greater or equal 3.7.0

Solution found at: https://github.com/cakephp/cakephp/issues/12682

ravage84 committed:
Allow failures in Travis with PHP 7.3

Because our dependency Gaufrette contains a PHP 7.3 incompatibility:

There was 1 error:
1) Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageBehaviorTest::testAfterDelete
"continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?
/home/travis/build/burzum/cakephp-file-storage/vendor/knplabs/gaufrette/src/Gaufrette/Util/Path.php:30
/home/travis/build/burzum/cakephp-file-storage/vendor/knplabs/gaufrette/src/Gaufrette/Adapter/Local.php:37
/home/travis/build/burzum/cakephp-file-storage/src/Storage/StorageManager.php:171
/home/travis/build/burzum/cakephp-file-storage/src/Storage/StorageManager.php:148
/home/travis/build/burzum/cakephp-file-storage/src/Storage/StorageManager.php:126
/home/travis/build/burzum/cakephp-file-storage/src/Storage/StorageTrait.php:44
/home/travis/build/burzum/cakephp-file-storage/src/Model/Behavior/FileStorageBehavior.php:161
/home/travis/build/burzum/cakephp-file-storage/tests/TestCase/Model/Behavior/FileStorageBehaviorTest.php:107

ravage84 committed:
Update Gaufrette to 0.8.1

This should fix the dreaded '"continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?' error.

 - knplabs/gaufrette updated from v0.7.0 to v0.8.1
   See changes: https://github.com/KnpLabs/Gaufrette/compare/v0.7.0...v0.8.1
   Release notes: https://github.com/KnpLabs/Gaufrette/releases/tag/v0.8.1

ravage84 committed:
Allow Gaufrette 0.7.x & 0.8.x side by side

We need to support from PHP 7.0 upwards and Gaufrette 0.8.x does not support PHP 7.0 anymore.

ravage84 committed:
Do not allow failures in Travis CI with PHP 7.3

We dependd on a compatible version of Gaufrette now.

ravage84 committed:
Try not to manually install/update composer & the dependencies

This currently doesn't work anyway.

ravage84 committed:
Fix CS

ravage84 commented:
@burzum I'm OK with it now. Can you have a look, please? //cc @liviascapin

burzum commented:
Awesome job @ravage84, thank you!

Do you want to wait for a review from @liviascapin as well?
